src/expression.c

```diff
@@ -9571,7 +9573,10 @@ Expression *DeleteExp::semantic(Scope *sc)
                 FuncDeclaration *fd = sd->dtor;
 
                 if (!f)
+                {
+                    semanticTypeInfo(sc, ts);
```

Comment:
We already do lower things like to reach in earlier semantic steps so it should be possible for other constructs as well.